x-pack/plugins/endpoint/server/services/endpoint/alert_query_builders.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { KibanaRequest } from 'kibana/server';
+import { EndpointAppConstants } from '../../../common/types';
+import { EndpointAppContext } from '../../types';
+
+export const kibanaRequestToAlertListQuery = async (
+  pagingProperties: Record<string, any>,
+  endpointAppContext: EndpointAppContext
+): Promise<Record<string, any>> => {
+  const DEFAULT_TOTAL_HITS = 10000;
+
+  // Calculate minimum total hits set to indicate there's a next page
+  const totalHitsMin = Math.max(
+    pagingProperties.fromIndex + pagingProperties.pageSize * 2,
+    DEFAULT_TOTAL_HITS
+  );
+
+  return {
+    body: {
+      track_total_hits: totalHitsMin,
+      query: {
+        match_all: {},
+      },
+      sort: [
+        {
+          '@timestamp': {
+            order: 'desc',
+          },
+        },
+      ],
+    },
+    from: pagingProperties.fromIndex,
+    size: pagingProperties.pageSize,
+    index: EndpointAppConstants.ALERT_INDEX_NAME,
+  };
+};
+
+export const getPagingProperties = async (
+  request: KibanaRequest<any, any, any>,
```

Comment:
Could we add types for `Params`, `Query`, and `Body` here?
```
KibanaRequest<Params, Query, Body>
```

Reply:
I need halp.